sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessSessionEventArgs.cs

```diff
@@ -85,5 +97,21 @@ public class ProcessSessionEventArgs : EventArgs
             BinaryData sessionState,
             CancellationToken cancellationToken = default) =>
             await _sessionReceiver.SetSessionStateAsync(sessionState, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Releases the session that is being processed. No receives will be initiated for the session and the
+        /// session will be closed.
+        /// The session may end up being reopened for processing immediately after closing if there are messages remaining in the session (
+        /// This depends on what other session messages may be in the queue or subscription).
+        /// </summary>
+        public virtual void ReleaseSession() =>
+            // manager will be null if instance created using the public constructor which is exposed for testing purposes
+            _manager?.CancelSession();
+
+        ///<inheritdoc cref="ServiceBusSessionReceiver.RenewSessionLockAsync(CancellationToken)"/>
+        public virtual async Task RenewSessionLockAsync(CancellationToken cancellationToken = default)
```

Comment:
Interesting...  I guess this is for potentially long-running initializations?

Reply:
Yeah, I think it makes sense to include as well. Basically these handlers should have all non-message operations.